docs/source/development/defining_new_models.md

```diff
@@ -193,47 +193,63 @@ def cleanup(self) -> None:
+So the example model used here would need to provide the file:
+`virtual_rainforest/models/freshwater/freshwater_schema.json`
 
-@register_schema("freshwater")
-def schema() -> dict:
-    """Defines the schema that the freshwater module configuration should conform to."""
+Writing JSONSchema documents can be very tedious. The following tools may be of use:
 
-    schema_file = Path(__file__).parent.resolve() / "freshwater_schema.json"
+* [https://www.jsonschema.net/app](https://www.jsonschema.net/app): this is a web
+  application that takes a data document - which is what the configuration file - and
+  automatically generates a JSON schema to validate it. You will need to then edit it
+  but you'll be starting with a valid schema!
+* [https://jsonschemalint.com/](https://jsonschemalint.com/) works the other way. It
+  takes a data document and a schema and checks whether the data is compliant. This can
+  be useful for checking errors.
 
-    with schema_file.open() as f:
-        config_schema = json.load(f)
+Both of those tools take data documents formatted as JSON as inputs, where we use TOML
+configuration files, but there are lots of web tools to convert TOML to JSON and back.
 
-    return config_schema
-```
+## Setting up the model `__init__.py` file
 
-## Ensuring that schema and models are always added to the registry
+All model directories need to include an `__init__.py` file. The simple presence of the
+`__init__.py` file tells Python that the directory content should be treated as module,
+but then the file needs to contain code to do two things:
 
-At the moment, a configuration schema only gets added to the schema registry when the
-model it belongs to is imported, and a `Model` class only gets added to the registry
-when the class itself is imported. This is a problem because the script that runs the
-main Virtual Rainforest simulation does not import these things directly. To circumvent
-this these imports needed to be placed in the top level `__init__.py` file (the one in
-the same folder as `main.py`). This won't pass the `pre-commit` checks unless `flake8`
-checks are turned off for the relevant lines. It's only strictly necessary to import the
-`Model` class, as this implicitly entails importing the specific model as a whole.
-However, for the sake of clarity we currently include both imports.
+1. The `__init__.py` file needs to register the JSONSchema file for the module. The
+  {meth}`~virtual_rainforest.core.config.register_schema` function takes a module name
+  and the path to the schema file and then, after checking the file can be loaded and is
+  valid, adds the schema to the schema registry
+  {data}`~virtual_rainforest.core.config.SCHEMA_REGISTRY`.
+1. It also needs to import the main BaseModel subclass. This is not going to be used
+   within the file, so needs `#noqa: 401` to surpress a `flake8` error. This step is
+   primarily to ensure that the subclass is always loaded from the same location but
```

Comment:
Not 100% sure I grasp the relevance of "loaded from the same location" isn't it more about ensuring that it's always loaded (so that `pytest` etc are guaranteed to find it)?

Reply:
It was opaque - I've given a bit more detail